# Human pancreas Smart-seq2 dataset 

```{r, echo=FALSE}
knitr::opts_chunk$set(cache=TRUE, error=FALSE, warning=FALSE, message=FALSE)
```

## Introduction

## Analysis code

### Data loading

```{r loading}
library(scRNAseq)
sce.seger <- SegerstolpePancreasData()
```

### Gene annotation

```{r gene-annotation}
library(AnnotationHub)
edb <- AnnotationHub()[["AH73881"]]
symbols <- rowData(sce.seger)$symbol
ens.id <- mapIds(edb, keys=symbols, keytype="SYMBOL", column="GENEID")
ens.id <- ifelse(is.na(ens.id), symbols, ens.id)

# Removing duplicated rows.
keep <- !duplicated(ens.id)
sce.seger <- sce.seger[keep,]
rownames(sce.seger) <- ens.id[keep]
```

### Sample annotation

```{r sample-annotation}
# We simplify the names of some of the relevant 
# column metadata fields for ease of access.
emtab.meta <- colData(sce.seger)[,c("cell type", 
    "individual", "single cell well quality")]
colnames(emtab.meta) <- c("CellType", "Donor", "Quality")
colData(sce.seger) <- emtab.meta

# Some editing of the cell type labels is necessary 
# for consistency with other data sets.
sce.seger$CellType <- gsub(" cell", "", sce.seger$CellType)
sce.seger$CellType <- paste0(
    toupper(substr(sce.seger$CellType, 1, 1)),
    substring(sce.seger$CellType, 2))
```

### Quality control

```{r quality-control}
# Removing low quality cells that were marked by the authors.
low.qual <- sce.seger$Quality == "low quality cell"

# Performing additional quality control, as some of the remaining 
# still have very low counts and numbers of detected features.
library(scater)
qcstats <- perCellQCMetrics(sce.seger)
filtered <- quickCellQC(qcstats, nmads=3,
    percent_subsets="altexps_ERCC_percent")
sce.seger <- sce.seger[,!(filtered$discard | low.qual)]
```

### Normalization

```{r normalization}
library(scran)
clusters <- quickCluster(sce.seger)
sce.seger <- computeSumFactors(sce.seger, clusters=clusters)
```

### Variance modelling

```{r variance-modelling}
# Not using cells with no spike-ins for variance modelling
# (Donor AZ has very spike-ins and is subsequently ignored).
for.hvg <- sce.seger[,librarySizeFactors(altExp(sce.seger)) > 0
    & sce.seger$Donor!="AZ"]
dec.seger <- modelGeneVarWithSpikes(for.hvg, "ERCC", block=for.hvg$Donor)
```
